### PR TITLE
Add more pgvector types.

### PR DESCRIPTION
**Pull request by vpetrovykh**
Add support for `sparsevec` and `halfvec`.
`halfvec` is handled as a float array (converted from float32 to float16 or vice versa as needed).
`sparsevec` is handled as a dict with non-zero values and "dimensions" specified in it.

### CODE REVIEW

**Comment by vpetrovykh** [comment]
The `halfvec` codecs need more robust conversion for float16. This seems to outline a good algo that I can implement here: http://www.fox-toolkit.org/ftp/fasthalffloatconversion.pdf.

Sometimes `_Float16` might be available via `#include <float.h>`, which should take care of casting between `float`/`double` and `float16`, but it would still need a fall-back.

File: edgedb/protocol/codecs/codecs.pyx [hunk 1]
```diff
@@ -798,6 +796,264 @@ cdef pgvector_decode(pgproto.CodecContext settings, FRBuffer *buf):
...
+        e -= 0x00800000         # Decrement exponent (1<<23)
+        m <<= 1                 # Shift mantissa
+
+    m &= ~0x00800000            # Clear leading 1 bit
+    e += 0x38800000             # Adjust bias ((127-14)<<23)
+
+    return m | e
+
+
+cdef generate_tables():
```
Comment (msullivan) [inline edgedb/protocol/codecs/codecs.pyx:832]: Could you put a comment in here (and the final commit message) with a cite to the algorithm you've implemented?

File: edgedb/protocol/codecs/codecs.pyx [hunk 2]
```diff
@@ -798,6 +796,264 @@ cdef pgvector_decode(pgproto.CodecContext settings, FRBuffer *buf):
...
+    return val
+
+
+cdef pgvector_sv_encode(pgproto.CodecContext settings, WriteBuffer buf,
+                        object obj):
+    cdef:
+        int32_t dim
+        Py_ssize_t nnz
+        # int32_t key
+        # float val
```
Comment (msullivan) [inline edgedb/protocol/codecs/codecs.pyx:985]: Dead

**Review by msullivan — APPROVED** [review]

File: edgedb/protocol/codecs/codecs.pyx (hunk 1 above)
Comment (vpetrovykh) [reply edgedb/protocol/codecs/codecs.pyx:840]: Comment added

File: edgedb/protocol/codecs/codecs.pyx (hunk 2 above)
Comment (vpetrovykh) [reply edgedb/protocol/codecs/codecs.pyx]: Cleaned up

**Pull request merged**